src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
+	
+	public AbstractDAO(BeanSession session) {
```

Comment:
Injection by setter is still working of course. Matter of taste, but I prefer injection by constructor because it is more "correct" since you must never have a dao without the beanSession. The setter does not enforce that, but the constructor does. I don't see as a big deal calling super(session). That said I do not mind moving back to setter if you want.